eng/common/scripts/update-docs-metadata.ps1

```diff
@@ -70,12 +70,15 @@ function GetAdjustedReadmeContent($pkgInfo, $lang){
     }
 
     $fileContent = $pkgInfo.ReadmeContent
-    $foundTitle = ""
 
     # only replace the version if the formatted header can be found
     $titleRegex = "(\# Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C))"
-    $fileContent = $pkgInfo.ReadmeContent -replace $titleRegex, "`${1} - Version $($pkgInfo.PackageVersion) `n"
-    
+    $fileTitle = ""
+    if ($pkgInfo.ReadmeContent -match $titleRegex) {
```

Comment:
While are you not using $fileContent instead of $pkgInfo.ReadmeContent directly? To ensure that we don't loose any processing we should use $fileContent everywhere after it is initialized at the top.

Reply:
Also is $fileContent a string or an array of strings it isn't clear based on usage here. Depending on that the -replace and -match might be slightly different.